pipelines/controller/src/main/java/org/openmrs/analytics/HiveTableManager.java

```diff
@@ -76,26 +75,75 @@ public void createResourceTables(
   private void createResourceTable(
       Connection connection, String resource, String timestamp, String thriftServerParquetPath)
       throws SQLException {
-    try (Statement statement = connection.createStatement()) {
-      String sql =
-          String.format(
-              "CREATE TABLE IF NOT EXISTS default.%s_%s USING PARQUET LOCATION '%s/%s/%s'",
-              resource,
-              timestamp,
-              THRIFT_CONTAINER_PARQUET_PATH_PREFIX,
-              thriftServerParquetPath,
-              resource);
-      statement.execute(sql);
 
-      // Drop canonical table if exists.
-      sql = String.format("DROP TABLE IF EXISTS default.%s", resource);
-      statement.execute(sql);
+    String sql =
+        String.format(
+            "CREATE TABLE IF NOT EXISTS default.%s_%s USING PARQUET LOCATION '%s/%s/%s'",
+            resource,
+            timestamp,
+            Constants.THRIFT_CONTAINER_PARQUET_PATH_PREFIX,
+            thriftServerParquetPath,
+            resource);
+    createResourceTable(connection, sql);
+
+    // Drop canonical table if exists.
+    sql = String.format("DROP TABLE IF EXISTS default.%s", resource);
+    createResourceTable(connection, sql);
+
+    // Create canonical table with latest parquet files.
+    sql =
+        String.format(
+            "CREATE TABLE IF NOT EXISTS default.%s USING PARQUET LOCATION '%s/%s/%s'",
+            resource,
+            Constants.THRIFT_CONTAINER_PARQUET_PATH_PREFIX,
+            thriftServerParquetPath,
+            resource);
+    createResourceTable(connection, sql);
+  }
+
+  /**
+   * The method creates resource tables with names suffixed with given timestamp if not present. It
+   * excepts complete parquet files path.
+   *
+   * @param resource
+   * @param timestamp
+   * @param thriftServerParquetPath
```

Comment:
nit: please either add description for params or remove `@param`s (I would prefer adding description as this is public but am fine either way).

Reply:
ack